test-integration/repay-loan/2_repay_loan_4_payments.js

```diff
@@ -0,0 +1,225 @@
+  assert.equal(
+    firstLoanStatusResult.toString(),
+    BigNumber((2).toString()),
+    'Invalid final loan staus.'
+  );
+
+  // Advance time to make first payment 30 days later 
+  console.log(`Advancing time to take out loan (current: ${(await timer.getCurrentDate())})...`);
+  const secondPaymentTimestamp = await timer.getCurrentTimestampInSecondsAndSum(daysToSeconds(15));
+  await timer.advanceBlockAtTime(secondPaymentTimestamp);
+
+  // Make 2nd payment
+  console.log('Making 2nd payment...');
+  await token.approve(lendingPoolInstance.address, payment, {from:borrower});
+  totalOwedResult = totalOwedResult.minus(payment);
+  const repay2_result = await loansInstance.repay(payment, lastLoanID, borrowerTxConfig);
+  // Check loan status
+  loans
+    .loanRepaid(repay2_result)
+    .emitted(lastLoanID, borrowerTxConfig.from, payment, borrowerTxConfig.from, totalOwedResult.toString());
+  const secondLoanStatus = await loansInstance.loans(lastLoanID) 
+  assert.equal(
+    secondLoanStatus.status.toString(),
+    loanStatuses.Active,
+    'Invalid #2 loan staus.'
+  );
+
+  // Advance time to make first payment 45 days later 
+  console.log(`Advancing time to take out loan (current: ${(await timer.getCurrentDate())})...`);
+  const thirdPaymentTimestamp = await timer.getCurrentTimestampInSecondsAndSum(daysToSeconds(15));
+  await timer.advanceBlockAtTime(thirdPaymentTimestamp);
+
+  // Make 3rd payment
+  console.log('Making 3rd payment...');
+  await token.approve(lendingPoolInstance.address, payment, {from:borrower});
+  const repay3_result = await loansInstance.repay(payment, lastLoanID, borrowerTxConfig);
+  // Check loan status
+  totalOwedResult = totalOwedResult.minus(payment);
+  loans
+    .loanRepaid(repay3_result)
+    .emitted(lastLoanID, borrowerTxConfig.from, payment, borrowerTxConfig.from, totalOwedResult.toString());
+  const thirdLoanStatus = await loansInstance.loans(lastLoanID) 
+  assert.equal(
+    thirdLoanStatus.status.toString(),
+    loanStatuses.Active,
+    'Invalid #3 loan staus.'
+  );
+
+  // Advance time to make first payment 55 days later 
+  console.log(`Advancing time to take out loan (current: ${(await timer.getCurrentDate())})...`);
+  const forthPaymentTimestamp = await timer.getCurrentTimestampInSecondsAndSum(daysToSeconds(10));
+  await timer.advanceBlockAtTime(forthPaymentTimestamp);
+
+  // Make 4th payment
+  console.log('Making final payment...');
+  await token.approve(lendingPoolInstance.address, payment, {from:borrower});
+  const repay4_result = await loansInstance.repay(payment, lastLoanID, borrowerTxConfig);
+
+  // Check final loan status 
+  totalOwedResult = totalOwedResult.minus(payment);
+  console.log('Expected (after 4# repay) total owed:  ', totalOwedResult.toString());
+  loans
+    .loanRepaid(repay4_result)
+    .emitted(lastLoanID, borrowerTxConfig.from, payment, borrowerTxConfig.from, totalOwedResult.toString());
+  const loanStatus = await loansInstance.loans(lastLoanID);
```

Comment:
Please, add assertions for the loanStatus.principalOwed == 0 and loanStatus.interestOwed == 0

Reply:
Added